sdcm/provision/scylla_yaml/scylla_yaml.py

```diff
@@ -147,6 +147,11 @@ def set_endpoint_snitch(cls, endpoint_snitch: str):
     internode_send_buff_size_in_bytes: int = None  # 0
     internode_recv_buff_size_in_bytes: int = None  # 0
     internode_compression: Literal['none', 'all', 'dc'] = None  # "none"
+    rpc_dict_training_when: Literal['always', 'never', 'when_leader'] = 'when_leader'
```

Comment:
I would recommend putting the actual defaults here (i.e. None)

and create an SCT configuration for specific test that would enabled it, for the sake of testing.
enabling it for all by default, should preferably done by scylla-core itself if decided as such.

see the `configurations` folder, and as example `configurations/tablets-initial-32.yaml` how a specific test can set it's own scylla.yaml parameters

Reply:
> I would recommend putting the actual defaults here (i.e. None)
> 
> and create an SCT configuration for specific test that would enabled it, for the sake of testing. enabling it for all by default, should preferably done by scylla-core itself if decided as such.

Why? The good thing about enabling it by default is that it gives more coverage with little effort. What's the drawback?

Reply:
the drawback is that we won't be testing the defaults from that point onwards.

if it's zero risk, change the default in scylla, if not the are reasons not to do so, they apply here in SCT as well.

Reply:
> I would recommend putting the actual defaults here (i.e. None)
> and create an SCT configuration for specific test that would enabled it, for the sake of testing.

@fruch 

In that case, would it be acceptable to enable dict training and internode compression for all *longevity* tests?

We can't enable this feature by default in Scylla because it has a performance cost. For some users, RPC compression is useless (e.g. because they have an on-premise deployment and they don't pay for the amount of data sent over network), and we don't want to regress them.

But we do want it to be stable and relatively cheap "by default". We might want to enable it by default in the cloud, because every cluster pays the network costs there.

> the drawback is that we won't be testing the defaults from that point onwards.

Internode compression only adds code paths, it doesn't remove them. So when you enable it, you are testing strictly more code. And I think that's a good thing?

Reply:
@fruch ^Ping.

Reply:
so you are saying it's not gonna be enabled by default for core, cause of fear of regression.
so it might cause regression on some of the tests

and for the "cheap" argument, every time we said something like that, we regretted it (see the enabling audit by default, we after a while got reverted)

I would still argue you do want specific case that can show different compression option vs. other options

and in this PR, you are not enabling the internode_compression at all, so it would only affect some cases using it.
which are not that many cases

Reply:
Anyhow, we should be trying it on a few cases before merging it
- one short longevity
- one case of rolling upgrades

as followup work after merge:
- one performance test - during operation(nemesis) one

Reply:
>  so it might cause regression on some of the tests

which is good, because it means we detected a bug, no?

Reply:
@kbr-scylla it means who owns the feature so do that leg work thinking/testing, not some other poor guy that happen to be trying do something else.

if we want to cover things, it should be systematic, not completely random.
people should be building test cases for their features. 

"lets enable all features all the time", kind of like let's default to RBNO,  and who cares about streaming from that point onwards....